controller/instance_manager_controller.go

```diff
@@ -265,6 +265,11 @@ func (imc *InstanceManagerController) syncInstanceManager(key string) (err error
 	if err != nil {
 		if datastore.ErrorIsNotFound(err) {
 			logrus.Infof("Longhorn instance manager %v has been deleted", key)
+			if err := imc.ds.DeletePDB(getPDBNameFromIMName(name)); err != nil {
```

Comment:
It's possible for us to miss this event. I don't think we should depend on the deletion notification. It's better to compare what we have set right now vs what's need to be set right now.

Reply:
I see. Thank you!

Reply:
Fixed. Thanks